data/data/bootstrap/files/usr/local/bin/release-image-download.sh.template

```diff
@@ -17,3 +17,16 @@ do
     echo "Pull failed. Retrying $RELEASE_IMAGE..."
 done
 
+# Sanity check the image metadata to see if the arches match
+image_arch=$(podman inspect $RELEASE_IMAGE --format {{ "{{.Architecture}}" }})
+host_arch=$(uname -m)
+case $host_arch in
+    "x86_64")  host_arch="amd64"   ;;
+    "aarch64") host_arch="arm64"   ;; # not used, just for completeness
+esac
+
+if [[ "$image_arch" != "$host_arch" ]]; then
+    echo "ERROR: release image arch $image_arch does not match host arch $host_arch"
+    exit 1
```

Comment:
A failing release-image.service won't prevent other services from starting. So is the purpose of this just to have a failing unit in the bootstrap gather?

Reply:
essential services like crio and kubelet won't start if `release-image.service` throws an error. so in that sense, we have caught this error early and let the user know that there is an arch mismatch in the specified release image.

Reply:
A release-image.service that fails will not prevent crio.service or kubelet.service from starting.

Reply:
ah..right ..yeah. but it does prevent crio-configure.service from starting which in turn disables bootkube.service when there is an error, so in that sense , this error is visible right away after running ` journalctl -b -f -u release-image.service -u bootkube.service`

Reply:
No, my point is that it *does not* prevent any of those services from starting.

Reply:
> No, my point is that it **does not** prevent any of those services from starting.

Ah, my bad. You are correct. Sorry for the noise. I looked at crio.service and kubelet.service but not bootkube.service.